src/silx/io/utils.py

```diff
@@ -32,13 +32,13 @@
 import sys
 import time
 import logging
-from typing import Generator
+from typing import Generator, Union, Optional
 import urllib.parse
 
 import numpy
 
 from silx.utils.proxy import Proxy
-import silx.io.url
+from silx.io.url import DataUrl
```

Comment:
Provided `silx.io` import `utils`, maybe relative import is more robust:
```suggestion
from .url import DataUrl
```